jetty-servlet/src/main/java/org/eclipse/jetty/servlet/FilterHolder.java

```diff
@@ -156,13 +161,41 @@ public void doStop()
 
     @Override
     public void destroyInstance(Object o)
-        throws Exception
     {
         if (o == null)
             return;
-        Filter f = (Filter)o;
-        f.destroy();
-        getServletHandler().destroyFilter(f);
+
+        Filter filter = (Filter)o;
+        // destroy the wrapped filter, in case there is special behaviour
+        filter.destroy();
+        // need to use the unwrapped filter because lifecycle callbacks such as
+        // postconstruct and predestroy are based off the classname and the wrapper
+        // classes are unknown outside the ServletHolder
+        getServletHandler().destroyFilter(unwrap(filter));
```

Comment:
The `ServletHolder` does these two operations in the opposite order.  However, I think the order here is correct, so the servlet holder needs to be changed to do the same order.